workshop/content/40-dotnet/30-hello-cdk/300-apigw.md

```diff
@@ -0,0 +1,194 @@
++++
+title = "API Gateway"
+weight = 300
++++
+
+Next step is to add an API Gateway in front of our function. API Gateway will
+expose a public HTTP endpoint that anyone on the internet can hit with an HTTP
+client such as [curl](https://curl.haxx.se/) or a web browser.
+
+We will use [Lambda proxy
+integration](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-create-api-as-simple-proxy-for-lambda.html)
+mounted to the root of the API. This means that any request to any URL path will
+be proxied directly to our Lambda function, and the response from the function
+will be returned back to the user.
+
+## Install the API Gateway construct library
+
+```
+dotnet add package Amazon.CDK.AWS.APIGateway
+```
+
+## Add a LambdaRestApi construct to your stack
+
+Going back to `lib/cdk-workshop-stack.ts`, let's define an API endpoint and associate it with our Lambda function:
+
+{{<highlight ts "hl_lines=3 19-23">}}
+using Amazon.CDK;
+using Amazon.CDK.AWS.Lambda;
+using Amazon.CDK.AWS.APIGateway;
+
+namespace CdkWorkshop
+{
+    public class CdkWorkshopStack : Stack
+    {
+        public CdkWorkshopStack(Construct parent, string id, IStackProps props) : base(parent, id, props)
```

Comment:
Make sure the `props` argument has a `= null` default (same here - updated template will have that).